File: 5Days-SoftwareArchitecture/typesript/Day2/Connascence/con_meaning_bad.js
var AccountServiceBad = /** @class */ (function () {
    function AccountServiceBad() {
    }
    AccountServiceBad.prototype.withdraw = function (amount) {
        return -1;
    };
    return AccountServiceBad;
}());
var ClientMeaningBad = /** @class */ (function () {
    function ClientMeaningBad() {
        this.svc = new AccountServiceBad();
    }
    ClientMeaningBad.prototype.attempt = function () {
        var result = this.svc.withdraw(500);
        if (result === -1)
            console.log("Insufficient funds (magic -1)");
    };
    return ClientMeaningBad;
}());
new ClientMeaningBad().attempt();

File: 5Days-SoftwareArchitecture/Java/Day 2/Connascence/CoN_Meaning_Bad.java
class AccountServiceBad {
    // return codes:
    // -1 = insufficient funds, 0 = success, 1 = error
    int withdraw(double amount) {
        return -1;
    }
}

class ClientMeaningBad {
    private AccountServiceBad service = new AccountServiceBad();

    void attempt() {
        int code = service.withdraw(500);
        if (code == -1) {
            System.out.println("Insufficient funds (magic -1)");
        }
    }
}

public class CoN_Meaning_Bad {
    public static void main(String[] args) {
        new ClientMeaningBad().attempt();
    }
}
